### PR TITLE
Animations refactoring

### PR DESCRIPTION
**Pull request by TeslaRus**
merging anf MInGW compile fix

### CODE REVIEW

File: src/skeletal_model.c [hunk 1]
```diff
@@ -822,14 +824,6 @@ int  Anim_SetNextFrame(struct ss_animation_s *ss_anim, float time)
     ss_anim->frame_time = (ss_anim->frame_time >= 0.0f) ? (ss_anim->frame_time) : (0.0f);
     ss_anim->frame_time += time;
     new_frame = ss_anim->frame_time / ss_anim->period;
-    if((ss_anim->current_animation == ss_anim->prev_animation) &&
```
Comment (TeslaRus) [inline src/skeletal_model.c]: I do not know what it was

Comment (TokyoSU) [reply src/skeletal_model.c]: i used the skeletal_model of this repo, so not by me :x

Comment (TeslaRus) [reply src/skeletal_model.c]: I know, it is my code, but I forgot why I had add that =)

**Comment by TokyoSU** [comment]
i fix src tomorrow.

File: src/skeletal_model.h [hunk 2]
```diff
@@ -78,12 +78,12 @@ typedef struct ss_bone_tag_s
 
     float                   qrotate[4];                                         // quaternion rotation
     
-#ifdef _WIN64
-	float                   local_transform[16];       // 4x4 OpenGL matrix for stack usage
-	float                   current_transform[16];    // 4x4 OpenGL matrix for global usage
-#elif __linux__
-	float                   local_transform[16] __attribute__((packed, aligned(16)));       // 4x4 OpenGL matrix for stack usage
-	float                   current_transform[16]  __attribute__((packed, aligned(16)));    // 4x4 OpenGL matrix for global usage
+#ifdef __GNUC__
```
Comment (TeslaRus) [inline src/skeletal_model.h]: more correct to check the compiler, because I use GCC under Windows, so It is possible to have __attribute__ under MSW

**Pull request merged**